locations/storefinders/amrest_eu.py

```diff
@@ -16,86 +16,91 @@ class AmrestEUSpider(Spider):
+        }
+        if self.api_source:
+            headers.update({"Source": self.api_source})
         for restaurant in response.json()["restaurants"]:
-            yield JsonRequest(
-                url=self.make_details_url(response.meta["root_url"], restaurant),
-                headers=self.base_headers | {"authorization": f'Bearer {response.meta["token"]}'},
-            )
-
-    def parse(self, response):
-        feature = response.json()["details"]
-
-        item = DictParser.parse(feature)
-
-        item["postcode"] = feature["addressPostalCode"]
-        item["housenumber"] = feature.get("addressStreetNo")
-        item["street"] = feature["addressStreet"]
-        item["opening_hours"] = self.parse_hours(feature["facilityOpenHours"])
-
-        apply_yes_no(Extras.DRIVE_THROUGH, item, feature.get("driveThru"))
-        apply_yes_no(Extras.DELIVERY, item, feature.get("delivery"))
-        apply_yes_no(Extras.TAKEAWAY, item, feature.get("takeaway"))
-
-        yield from self.parse_item(item, feature)
-
-    def parse_item(self, item, feature, **kwargs):
+            restaurant_id = restaurant["id"]
+            if self.api_channel:
+                yield JsonRequest(
+                    url=f"https://api.amrest.eu/amdv/ordering-api/{self.api_brand_country_key}/rest/v2/restaurants/{restaurant_id}/{self.api_channel}",
+                    headers=headers,
+                    callback=self.parse_restaurant_details,
+                )
+            else:
+                yield JsonRequest(
+                    url=f"https://api.amrest.eu/amdv/ordering-api/{self.api_brand_country_key}/rest/v2/restaurants/details/{restaurant_id}",
+                    headers=headers,
+                    callback=self.parse_restaurant_details,
+                )
+
+    def parse_restaurant_details(self, response):
+        location = response.json()["details"]
+        item = DictParser.parse(location)
+        item["ref"] = str(location["id"])
+        item["postcode"] = location["addressPostalCode"]
+        item["housenumber"] = location.get("addressStreetNo")
+        item["street"] = location["addressStreet"]
+        item["opening_hours"] = self.parse_hours(location["facilityOpenHours"])
+
+        apply_yes_no(Extras.DELIVERY, item, location.get("delivery"), False)
+        apply_yes_no(Extras.TAKEAWAY, item, location.get("takeaway"), False)
+        apply_yes_no(Extras.INDOOR_SEATING, item, location.get("dineIn"), False)
+        extra_features = [feature["key"] for feature in location.get("filters", {})]
+        if extra_features:
+            apply_yes_no(Extras.DRIVE_THROUGH, item, "driveThru" in extra_features, False)
+            apply_yes_no(Extras.OUTDOOR_SEATING, item, location.get("garden"), False)
+            apply_yes_no(Extras.WIFI, item, "wifi" in extra_features, False)
+
+        yield from self.parse_item(item, location)
+
+    def parse_item(self, item: Feature, location: dict):
         yield item
```

Comment:
Didn't we settle on `post_process_item()`? Or is this semantically different?

Reply:
I think the balance was post_process_item. I'll adjust this, I was taking @davidhicks' work verbatim for the short term; to get the autospidergen branch mergable :)